src/explore-education-statistics-admin/src/pages/release/edit-release/content/ReleaseContext.tsx

```diff
@@ -0,0 +1,237 @@
+        if (
+          draft.release === undefined ||
+          typeof draft.release[sectionKey] !== 'object'
+        ) {
+          throw new Error('REMOVE_BLOCK_FROM_SECTION: failed');
+        } else if (sectionKey === 'content') {
+          draft.release[sectionKey] = draft.release[sectionKey].map(section => {
+            if (section.id === sectionId) {
+              return {
+                ...section,
+                content: section.content?.filter(
+                  contentBlock => (contentBlock.id as string) !== blockId,
+                ),
+              } as ContentSection<EditableContentBlock>;
+            }
+            return section;
+          });
+        } else {
+          (draft.release[sectionKey] as ContentSection<
+            EditableContentBlock
+          >).content = (draft.release[sectionKey] as ContentSection<
+            EditableContentBlock
+          >).content?.filter(
+            contentBlock => (contentBlock.id as string) !== blockId,
+          );
+        }
+      });
+    case 'UPDATE_BLOCK_FROM_SECTION': {
+      return produce<State>(state, draft => {
+        const { block, meta } = action.payload;
+        const { sectionId, blockId, sectionKey } = meta;
+        if (
+          draft.release === undefined ||
+          typeof draft.release[sectionKey] !== 'object'
+        ) {
+          throw new Error('UPDATE_BLOCK_FROM_SECTION: failed');
+        } else if (sectionKey === 'content') {
+          draft.release[sectionKey] = draft.release[sectionKey].map(section => {
+            if (section.id === sectionId) {
+              return {
+                ...section,
+                content: section.content?.map(contentBlock => {
+                  if ((contentBlock.id as string) === blockId) {
+                    return block;
+                  }
+                  return contentBlock;
+                }),
+              } as ContentSection<EditableContentBlock>;
+            }
+            return section;
+          });
+        } else {
+          (draft.release[sectionKey] as ContentSection<
+            EditableContentBlock
+          >).content = (draft.release[sectionKey] as ContentSection<
+            EditableContentBlock
+          >).content?.map(contentBlock => {
+            if ((contentBlock.id as string) === blockId) {
+              return block;
+            }
+            return contentBlock;
+          });
+        }
+      });
+    }
+    case 'ADD_BLOCK_TO_SECTION': {
+      return produce<State>(state, draft => {
+        const { block, meta } = action.payload;
+        const { sectionId, sectionKey } = meta;
+        if (
+          draft.release === undefined ||
+          typeof draft.release[sectionKey] !== 'object'
+        ) {
+          throw new Error('ADD_BLOCK_TO_SECTION: failed');
+        } else if (sectionKey === 'content') {
+          draft.release[sectionKey] = draft.release[sectionKey].map(section => {
```

Comment:
Take the suggestions from above to write the rest of this branch in a more Immer idiomatic way.